### PR TITLE
Hasten CI

### LINKED ISSUE #152: Decrease CI time to run tests
It can be up to an hour (or worse: cancelled) to run the main tests. It should be less than 10 minutes, ideally only a few.

Move to Circle CI or pay GitHub.

### PR DESCRIPTION
**Pull request by Adaickalavan**
+ Original CI duration: about 35 minutes
+ Current CI duration: about 15 minutes
+ Changes: CI config file now uses strategy matrix which helps run test cases in parallel.
+ Longest individual test case `/smarts/env/tests/test_rllib_hiway_env.py` takes a minimum raw time of 10 minutes to run (excluding setup time)

Closes #152 .

### CODE REVIEW

**Comment by liamchzh** [comment]
Very impressive and worth doing such an enhancement before we migrate to self-host workers or any paid CI.

**Review by liamchzh — APPROVED** [review]

**Review by Gamenot — APPROVED** [review]
Thanks for being thorough with this.

**Comment by Gamenot** [comment]
@Adaickalavan I think this is enough time spent on this problem. This should go in once the test-formatting is addressed.

File: smarts/core/remote_agent_buffer.py [hunk 1]
```diff
@@ -197,8 +197,6 @@ def get_manager_channel_stub(addr):
         # Wait until the grpc server is ready or timeout after 30 seconds
         grpc.channel_ready_future(channel).result(timeout=30)
     except grpc.FutureTimeoutError:
-        raise RemoteAgentException(
```
Comment (liamchzh) [inline smarts/core/remote_agent_buffer.py:200]: :+1:

**Review by liamchzh — COMMENTED** [review]
Is the caching buying us more time? :smile:

**Pull request merged**